product_template_websale_search_extended/controllers/main.py

```diff
@@ -0,0 +1,14 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import fields, http, tools, _
+from odoo.http import request
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSaleSearch():
+
+    def _add_search_domain(domain, search, field):
+        domain.insert(2, '|')
+        domain += [(field, 'ilike', search)]
```

Comment:
You're trying to add an OR expression, a more secure way is using `expression.OR` tool, like this:

https://github.com/OCA/account-financial-tools/blob/b964802aab884382c90966c50bef4c7ad361dd41/account_move_fiscal_year/models/account_move.py#L36-L43

Could you try?